as/itests/org.jboss.tools.as.ui.bot.itests/src/org/jboss/tools/as/ui/bot/itests/parametized/server/ShowInContextMenuTest.java

```diff
@@ -60,36 +69,205 @@ public void cleanUp() {
+		assertTrue("Web Management Console is missing from the context menu", isInContextMenu());
 	} 
+	
+	//Options do what are they supposed to do
+	
 	@Test
 	public void showInWebBrowserOpensProperPage() throws InterruptedException {
+		startAndSelectServer();
+		
+		selectItem("Web Browser");
+		
+		testBrowser();
+	}
+	
+	@Test
+	public void showInWebManagementConsoleOpensProperPage() throws InterruptedException {
+		startAndSelectServer();
+		
+		selectItem("Web Management Console");
+		
+		testBrowser();
+	}
+	
+	@Test
+	public void showInConsoleOpensProperPage() {
+		startAndSelectServer();
+		
+		selectItem("Console");
+		
+		ConsoleView cv = new ConsoleView();
+		
+		assertTrue("Console does not contain proper text: It has " + cv.getConsoleText(),
+				cv.getConsoleText().contains("wildfly"));
+	}
+	
+	@Test
+	public void showInDebugOpensProperPage() {
+		startAndSelectServer();
+		
+		selectItem("Debug");
+		
+		sv.open();
+		server.select();
+		
+		selectItem("Debug");
+		
+		LaunchView dv = new LaunchView();
+			
+		boolean containsWildfly = false;
+		
+		for (TreeItem item : dv.getSelectedItem().getParent().getAllItems()) {
+			if (item.getText().contains("WildFly")) {
+				containsWildfly = true;
+				break;
+			}
+		}
+		
+		assertTrue("Debug View does not contain WildFly server", containsWildfly);
+	}
+	
+	@Test
+	public void showInJMXNavigatorOpensProperPage() {
+		startAndSelectServer();
+		
+		selectItem("JMX Navigator");
+		
+		JMXNavigatorView jmx = new JMXNavigatorView();
+		
+		JMXConnectionItem jmxConnectionsItem = jmx.getServerConnectionsItem();
+		
+		boolean containsWildfly = false;
+		
+		for (JMXConnection connection : jmxConnectionsItem.getConnections()) {
+			if (connection.getLabel().getName().contains("WildFly")){
+				containsWildfly = true;
+				break;
+			}
+		}
+		
+		assertTrue("JMX Navigator does not contain WildFly server", containsWildfly);
+	}
+	
+	//Methods for simpler code
+	
+	private void startAndSelectServer() {
 		if(!server.getLabel().getState().isRunningState()) {
 			server.start();
 		}
 		
 		sv.open();
 		server = sv.getServer("WildFly 24+ Server");
+	}
+	
+	private void selectItem(String item) {
+		new ContextMenu().getItem("Show In",item).select();
+	}
+	
+	private boolean isEnabled(String item) {
+		return new ContextMenu().getItem("Show In", item).isEnabled();
+	}
+	
+	private String errorMsg(String option, boolean state) {
+		String status = (state) ? "active" : "inactive";
+		String reverseStatus = (state) ? "inactive" : "active";
+		
+		return option + " option in Show in context menu is " + status + ". Should be " + reverseStatus + ".";
+	}
+	
+	private boolean isInContextMenu() {
+		List<MenuItem> items = new ContextMenu().getItem("Show In").getChildItems();
+		boolean isInMenu = false;
 		
-		new ContextMenu().getItem("Show In", "Web Browser").select();
+		for (MenuItem item : items) {
+			if (item.getText().contains("Web Management Console")) {
+				isInMenu = true;
```

Comment:
Put only a return statement here in order to not iterate over every item in the list. you need only first fulfilled condition.